src/main/java/mate/academy/book/store/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package mate.academy.book.store.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.book.store.dto.shoppingcart.request.CreateBookItemDto;
+import mate.academy.book.store.dto.shoppingcart.request.UpdateBookQuantityDto;
+import mate.academy.book.store.dto.shoppingcart.response.ShoppingCartResponseDto;
+import mate.academy.book.store.mapper.user.ShoppingCartMapper;
+import mate.academy.book.store.model.CartItem;
+import mate.academy.book.store.model.ShoppingCart;
+import mate.academy.book.store.model.User;
+import mate.academy.book.store.repository.shoppingcart.ShoppingCartRepository;
+import mate.academy.book.store.service.CartItemService;
+import mate.academy.book.store.service.ShoppingCartService;
+import mate.academy.book.store.service.UserService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemService cartItemService;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final UserService userService;
+
+    @Override
+    public ShoppingCartResponseDto getUserCart() {
+        User user = userService.getAuthenticatedUser();
+        return shoppingCartMapper
+                .toDto(shoppingCartRepository.findShoppingCartByUserId(user.getId()));
+    }
+
+    @Override
+    public ShoppingCartResponseDto addBookToUserCart(CreateBookItemDto requestDto) {
+        ShoppingCart shoppingCart = getShoppingCartByUser();
+        CartItem cartItem = cartItemService.save(requestDto);
+        shoppingCart.getCartItems().add(cartItem);
+        return getUserCart();
+    }
```

Comment:
I'd recommend to add logic checking if such book is already in a cart.
It's rather common situation when user just click "buy" button twice and i'd not expect that frontend will handle this situation and convert to quantity change request

Reply:
i already check this case : here when i create the table with items :               
                - column:
                  name: book_id
                  type: bigint
                  constraints:
                    nullable: false
                    unique: true